code/game/objects/structures/lattice.dm

```diff
@@ -72,3 +72,12 @@
 		C.deconstruct()
 	..()
 
+
+/obj/structure/lattice/catwalk/lava
+	name = "thermal shielded  catwalk"
+	desc = "A reinforced catwalk with thermal shielding for protection from hostile terrain."
+	icon_state = "miningcatwalk"
+
+/obj/structure/lattice/catwalk/lava/attackby(obj/item/C, mob/user, params)
```

Comment:
this prevents you from unwelding it, intentional? and if so, make that a bool var on the base catwalk and check for it in the base catwalk ```attackby()```, then also set it on swarmer catwalks please.